drivers/flash/flash_npcx_fiu_nor.c

```diff
@@ -273,31 +274,34 @@ static int flash_npcx_nor_erase(const struct device *dev, off_t addr,
 	}
 
 	/* size must be a multiple of sectors */
-	if ((size % config->min_erase_size) != 0) {
+	if ((size % BLOCK_4K_SIZE) != 0) {
 		LOG_ERR("Size %d is not a multiple of sectors", size);
 		return -EINVAL;
 	}
 
+	offset_64k_aligned = (addr % BLOCK_64K_SIZE) == 0;
+	offset_4k_aligned = (addr % BLOCK_4K_SIZE) == 0;
+
 	/* Select erase opcode by size */
 	if (size == config->flash_size) {
 		flash_npcx_uma_cmd_only(dev, SPI_NOR_CMD_WREN);
 		/* Send chip erase command */
 		flash_npcx_uma_cmd_only(dev, SPI_NOR_CMD_CE);
 		return flash_npcx_nor_wait_until_ready(dev);
-	} else if (config->min_erase_size == KB(4)) {
-		opcode = SPI_NOR_CMD_SE;
-	} else if (config->min_erase_size == KB(64)) {
-		opcode = SPI_NOR_CMD_BE;
-	} else {
-		return -EINVAL;
 	}
 
 	while (size > 0) {
 		flash_npcx_uma_cmd_only(dev, SPI_NOR_CMD_WREN);
 		/* Send page/block erase command with addr */
-		flash_npcx_uma_cmd_by_addr(dev, opcode, addr);
-		addr += config->min_erase_size;
-		size -= config->min_erase_size;
+		if ((size >= BLOCK_64K_SIZE) && offset_64k_aligned) {
+			flash_npcx_uma_cmd_by_addr(dev, SPI_NOR_CMD_BE, addr);
+			addr += BLOCK_64K_SIZE;
+			size -= BLOCK_64K_SIZE;
+		} else if ((size >= BLOCK_4K_SIZE) && offset_4k_aligned) {
+			flash_npcx_uma_cmd_by_addr(dev, SPI_NOR_CMD_SE, addr);
+			addr += BLOCK_4K_SIZE;
+			size -= BLOCK_4K_SIZE;
+		}
```

Comment:
This could use an "else" case, I understand it should not be possible but just in case the code gets changed and something odd happens.

Reply:
I added an else which will return error, there is no opcode for erasing 1 byte hence went with it